module/apmgoredisv8/hook.go

```diff
@@ -0,0 +1,96 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build go1.11
+
+package apmgoredisv8
+
+import (
+	"bytes"
+	"context"
+	"strings"
+
+	"github.com/go-redis/redis/v8"
+
+	"go.elastic.co/apm"
+)
+
+// goRedisApmHook is an implementation of redis.Hook that reports cmds as spans to Elastic APM.
+type goRedisApmHook struct{}
+
+// NewGoRedisApmHook returns a redis.Hook that reports cmds as spans to Elastic APM.
+func NewGoRedisApmHook() redis.Hook {
+	return &goRedisApmHook{}
+}
```

Comment:
```suggestion
// hook is an implementation of redis.Hook that reports cmds as spans to Elastic APM.
type hook struct{}

// NewHook returns a redis.Hook that reports cmds as spans to Elastic APM.
func NewGoRedisApmHook() redis.Hook {
	return &hook{}
}
```

The package name provides the context of what type of hook it is, so having it in the function name stutters. Calling it "NewHook" is clear: `apmgoredisv8.NewHook()`.